File: sample-java/src/main/java/com/mapp/engagesample/MyPushBroadcast.java
package com.mapp.engagesample;

import android.util.Log;

import androidx.annotation.NonNull;

import com.appoxee.shared.LocalPushBroadcast;
import com.appoxee.shared.MappPush;

public class MyPushBroadcast extends LocalPushBroadcast {

    private void print(String action, MappPush push) {
        Log.d("MyPushBroadcast", "Action: " + action + "; Push: " + push);
    }

    @Override
    public void onReceived(@NonNull MappPush push) {
        print("onReceived", push);
    }

    @Override
    public void onOpened(@NonNull MappPush push) {
        print("onOpened", push);
    }

    @Override
    public void onSilent(@NonNull MappPush push) {
        print("onSilent", push);
    }

    @Override
    public void onDismissed(@NonNull MappPush push) {
        print("onDismissed", push);
    }

    @Override
    public void onButtonClick(@NonNull MappPush push) {
        print("onButtonClick", push);
    }

    @Override
    public void onRichPush(@NonNull MappPush push) {
        print("onRichPush", push);
    }
}

File: sample-kotlin/bin/src/main/java/com/mapp/engagesample/MyPushBroadcast.kt
package com.mapp.engagesample

import android.util.Log
import com.appoxee.shared.LocalPushBroadcast
import com.appoxee.shared.MappPush

class MyPushBroadcast : LocalPushBroadcast() {
    private fun log(action: String, push: MappPush) {
        val title = push.title
        val content = push.content
        Log.i("MyPushBroadcast", "ACTION: $action; PUSH: $push; TITLE: $title; CONTENT: $content")
    }

    override fun onReceived(push: MappPush) {
        log("onReceived", push)
    }

    override fun onOpened(push: MappPush) {
        log("onOpened", push)
    }

    override fun onSilent(push: MappPush) {
        log("onReceived", push)
    }

    override fun onDismissed(push: MappPush) {
        log("onDismissed", push)
    }

    override fun onButtonClick(
        push: MappPush
    ) {
        log("onButtonClick", push)
    }

    override fun onRichPush(push: MappPush) {
        log("onRichPush", push)
    }
}